pkg/mimirtool/config/cortex.go

```diff
@@ -1065,6 +1065,13 @@ var removedConfigPaths = append(gemRemovedConfigPath, []string{
 	"ruler.sharding_strategy",                // -ruler.sharding-strategy
 	"store_gateway.sharding_enabled",         // -store-gateway.sharding-enabled
 	"store_gateway.sharding_strategy",        // -store-gateway.sharding-strategy
+
+	// Removed in 2.1, 2.2 and 2.3
+	"distributor.extend_writes",                          // -distributor.extend-writes
+	"ingester.join_after",                                // -ingester.join-after; is deprecated and will be removed in 2.4, but we can remove it now
+	"querier.shuffle_sharding_ingesters_lookback_period", // -querier.shuffle-sharding-ingesters-lookback-period
+	"ruler.flush-period",                                 // -ruler.flush-period
```

Comment:
```suggestion
	"ruler.flush_period",                                 // -ruler.flush-period
```

Reply:
I decided to add tests that check that these flags are ok because this is the second we catch a mistake. Turns out that most of the flags we had for GEM were actually wrong. Also cortex was missing some flags that we had to manually add (in descriptors/cortex-v1.11.0-flags-only.json), so I updated these too.